python/lsst/ts/aos/utils/m1m3/M1M3LUT.py

```diff
@@ -145,52 +164,80 @@ async def update_lut_force_balance(
-        new_poly = Polynomial.fit(
-            90 - elevations, forces_resampled[query_forces[idx]], polynomial_degree
+    for row in M1M3FATable.FATABLE:
+        idx = row.get_index(index_type)
+        if idx is None:
+            continue
+
+        new_poly, statistics = Polynomial.fit(
+            90 - elevations,
+            forces_resampled[query_forces[idx]],
+            polynomial_degree,
+            full=True,
         )
+        res = statistics[0][0]
+        print(f"Fit {row.actuator_id} {axis} ({idx: 3}) residuals {res:.5f}")
+        residuals.append(res)
+
         coefs = np.flip(new_poly.convert().coef)
 
-        actuator_id = ids[np.where(axis_indices == idx)[0][0]]
         if force_type == ForceType.BALANCE:
             coefs = np.insert(coefs, 0, 0)
-            table_file.loc[table_file["ID"] == actuator_id] += coefs
+            table_file.loc[table_file["ID"] == row.actuator_id] += coefs
         elif force_type == ForceType.APPLIED:
-            coefs = np.insert(coefs, 0, actuator_id)
-            table_file.loc[table_file["ID"] == actuator_id] = coefs
+            coefs = np.insert(coefs, 0, row.actuator_id)
+            table_file.loc[table_file["ID"] == row.actuator_id] = coefs
+
+    print(
+        f"Residuals {np.min(residuals):.5f} {np.mean(residuals):.5f} {np.max(residuals):.5f}"
+    )
 
     # Save the updated LUT file
-    print("Saving LUT file...")
-    table_file.to_csv(f"Elevation{axis}Table.csv", index=False)
+    print(f"Saving LUT file {out_file}")
+    table_file.to_csv(out_file, index=False)
 
 
 def parse_arguments() -> argparse.Namespace:
     """Parse command line arguments."""
     parser = argparse.ArgumentParser(
-        description="Update LUT file with force balance data"
+        description="Update M1M3 LUT file with force balance data"
     )
     parser.add_argument(
         "force_type", choices=["Balance", "Applied"], help="Type of forces to be used"
     )
     parser.add_argument(
-        "start_time", help="Start time in a valid format: 'YYYY-MM-DD HH:MM:SSZ'"
+        "start_time",
+        help="Start time in a valid format: 'YYYY-MM-DD HH:MM:SSZ'",
+        type=Time,
     )
     parser.add_argument(
-        "end_time", help="End time in a valid format: 'YYYY-MM-DD HH:MM:SSZ'"
+        "end_time", help="End time in a valid format: 'YYYY-MM-DD HH:MM:SSZ'", type=Time
     )
     parser.add_argument(
-        "axis", choices=["X", "Y", "Z"], help="Axis of the force balance to be updated"
+        "--axes",
+        choices=["X", "Y", "Z", "XY", "XZ", "YZ", "XZY"],
+        default="XYZ",
+        help="Axis of the force balance to be updated. Defaults to all axis.",
     )
 
-    lut_path = f"{os.environ['HOME']}" "/develop/ts_m1m3support/SettingFiles/Tables/"
     parser.add_argument(
-        "--lut_path", default=lut_path, help="Path to the LUT file to be updated"
+        "--lut_path", default=None, help="Path to the LUT file to be updated"
```

Comment:
I think there are two options. Since you made that if lut_path is None we create an empty LUT, then I would say lut_path should be required and not None for the "Balance" approach. However, it should not be required for the "Applied" approach.

Reply:
There is now asset for Balance and printout if the force is missing.